packages/block-editor/src/components/inserter/block-list.js

```diff
@@ -55,64 +41,26 @@ export function InserterBlockList( {
 	rootClientId,
 	onInsert,
 	onHover,
-	__experimentalSelectBlockOnInsert: selectBlockOnInsert,
 	filterValue,
 	debouncedSpeak,
 } ) {
-	const {
-		categories,
-		collections,
-		items,
-		rootChildBlocks,
-		fetchReusableBlocks,
-	} = useSelect(
+	const [ items, categories, collections, onSelectItem ] = useBlockTypesState(
+		rootClientId,
+		onInsert
+	);
+	const { rootChildBlocks } = useSelect(
```

Comment:
You can avoid destructuring here and just return the value to assign to `rootChildBlocks` directly, rather than as the single-property in an object.

Reply:
I actually did this on purpose thinking we might need to add other selectors later, I  don't mind the change though.